controllers/mariadbdatabase_controller.go

```
@@ -161,6 +197,11 @@ func (r *MariaDBDatabaseReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		serviceAccount = dbMariadb.RbacResourceName()
 	}
 
+	instance.Status.Conditions.MarkTrue(
```

You can set the MariaDBServerReadyCondition to False with a message at L180 and L190 to indicate that the DB bootstrap is still ongoing. So the CR status will show the progress.

---

Also is case of error from `r.getDatabaseObject()` you can report that error in this condition too

---

So that should happen in mariadbaccount controller also, which has the same code.  right now the logic is that it shows "unknown".

---

I added this to all four spots (database, account, both for galera and mariadb )

---

which is now two spots now that mariadb CR removal has landed